src/MongoDB/WriteResult.c

```diff
@@ -427,13 +428,19 @@ static HashTable* php_phongo_writeresult_get_debug_info(zval* object, int* is_te
 		state.map.document_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
 
 		bson_iter_array(&iter, &len, &data);
-		php_phongo_bson_to_zval_ex(data, len, &state);
+		if (php_phongo_bson_to_zval_ex(data, len, &state)) {
+			hasUpsertedIds = true;
 #if PHP_VERSION_ID >= 70000
-		ADD_ASSOC_ZVAL_EX(&retval, "upsertedIds", &state.zchild);
+			ADD_ASSOC_ZVAL_EX(&retval, "upsertedIds", &state.zchild);
 #else
-		ADD_ASSOC_ZVAL_EX(&retval, "upsertedIds", state.zchild);
+			ADD_ASSOC_ZVAL_EX(&retval, "upsertedIds", state.zchild);
 #endif
-	} else {
+		} else {
+			zval_ptr_dtor(&state.zchild);
+		}
+	}
+
+	if (!hasUpsertedIds) {
```

Comment:
If I understand correctly, this just avoids populating the "upsertedIds" array if we know an exception was thrown, correct? Was this only introduced because we refactored to get rid of the large `else` block?

No problem if that's the case (the less `else` blocks the better); I just want to make sure.

Reply:
The refactoring was necessary to add the empty value to the array, which is obviously useless in the case of an exception (as you've noted before in this review). I've changed it back for now to avoid refactoring this without a functional need, but we may revisit this later.